app/models/post.rb

```diff
@@ -1,2 +1,12 @@
 class Post < ApplicationRecord
+  belongs_to :users
+  has_many :likes, :comments
+
+  def update_post_counter
+    users.posts_counter += 1
+  end
+
```

Comment:
[OPTIONAL]

- It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, by this you can make the `update_post_counter` method private which will increase the security.

```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    user.increment!(:posts_counter)
  end
```

Reply:
thanks for  this.. will use it when i understand it best.